Controllers/OperationController.cs

```diff
@@ -146,5 +152,18 @@ public async Task<HttpResponseMessage> Invoke(OperationInfo info)
 
             return await _armRepository.InvokeAsync(Request, executeRequest);
         }
+
+        private static bool IsValidHost()
+        {
+            if (HttpContext.Current.Request.Url.IsLoopback)
+            {
+                return true;
+            }
+
+            // For Azure scenarios we do extra checks for cross domains
+            var requestHost = HttpContext.Current.Request.UrlReferrer?.Host;
+            return string.Equals(requestHost, "resources.azure.com", StringComparison.OrdinalIgnoreCase) ||
+                   string.Equals(requestHost, "resources-staging.azure.com", StringComparison.OrdinalIgnoreCase);
+        }
```

Comment:
I would just do endswith ".azure.com" - this domain we own and give us flexibility in the future.